### PR TITLE
[v7r0] MySQL LIKE Query

### PR DESCRIPTION
**Pull request by dhrjklt**
Ability to perform LIKE queries using DIRAC native MySQL API. 
BEGINRELEASENOTES
*Core
CHANGE: Adding MySQL LIKE query feature.

ENDRELEASENOTES

### CODE REVIEW

**Review by fstagni — DISMISSED** [review]

**Comment by chaen** [comment]
What's the use case exactly ? Somehow, this makes me feel unsafe  from the injection POV. Rather than parsing with regex, I'd add an extra parameter to the call, explicitly for parameters requiring a `LIKE` statement. This would also limit the places where *mistakes* can happen and DDOS the DB...

**Review by chrisburr — CHANGES_REQUESTED** [review]
I agree with Chris. I'm fairly sure this would cause LFNs like `/lhcb/user/c/cburr/2016_first_50%.root` to match lots of entries that aren't intended. The same would apply to job names, groups, minor/application status and probably more values.

**Comment by fstagni** [comment]
Dear @dhrjklt would you please consider the comments above?

**Comment by dhrjklt** [comment]
Dear @fstagni  sorry for the late reply. I'm working on it.

File: Core/Utilities/MySQL.py [hunk 1]
```diff
@@ -151,6 +151,7 @@
 import time
 import threading
 import MySQLdb
+import re
```
Comment (chrisburr) [inline Core/Utilities/MySQL.py:154]: I think this is no longer used:
```suggestion
```

Comment (dhrjklt) [reply Core/Utilities/MySQL.py]: sorry, fixed!

File: Core/Utilities/MySQL.py [hunk 2]
```diff
@@ -1076,7 +1074,7 @@ def getDistinctAttributeValues(self, table, attribute, condDict=None, older=None
 #############################################################################
   def buildCondition(self, condDict=None, older=None, newer=None,
                      timeStamp=None, orderAttribute=None, limit=False,
-                     greater=None, smaller=None, offset=None):
+                     greater=None, smaller=None, offset=None, likeQuery=False):
```
Comment (hmiyake) [inline Core/Utilities/MySQL.py:1076]: You may add explanation of the switch in docstring

**Comment by dhrjklt** [comment]
@fstagni and @chrisburr please let me know if the changes are okay?

**Comment by chaen** [comment]
Seems ok to me

**Comment by chrisburr** [comment]
Seem okay 👍

Minor comment: Perhaps `likeQuery` should be `useLikeQuery` to make it clear it's a boolean (naively reading the name I would expect it to be a string).

**Review by fstagni — APPROVED** [review]

**Comment by dhrjklt** [comment]
@chrisburr fixed!

**Comment by atsareg** [comment]
Is there a particular reason to make it against v6r22 branch ? Can it go to v7r1 instead ?

**Comment by fstagni** [comment]
I think the reason is that this is requested by Belle2 who is still on v6r22. TBH I would bump it to at least v7r0.

**Comment by dhrjklt** [comment]
Dear @atsareg  and @fstagni,

Sorry for my late reply. I just confirmed with Belle2 and they confirmed that they are leaving v6r22 and v7r0 is preferable.

**Review by atsareg — APPROVED** [review]

**Pull request merged**